src/Particle/MCSample.cpp

```diff
@@ -0,0 +1,30 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2020  QMCPACK developers.
+//
+// File developed by: Mark Dewing, mdewing@anl.gov, Argonne National Laboratory
+//
+// File created by: Mark Dewing, mdewing@anl.gov, Argonne National Laboratory
+//////////////////////////////////////////////////////////////////////////////////////
+
+/** @file MCSample.cpp
+ * @brief Stores particle configurations for later use in DMC and wavefunction optimization
+ *
+ * Stores less temporary data than the buffer object.
+ */
+
+#include "Particle/MCSample.h"
+
+namespace qmcplusplus
+{
+using Walker_t = ParticleSet::Walker_t;
+
+MCSample convertWalkerToSample(Walker_t& walker)
+{
+  return MCSample(walker);
```

Comment:
Could you explain why this function is needed? When convertWalkerToSample is used, the developer knows it is constructing a sample from a walker. So why not just using MCSample(walker) in place instead of this function?